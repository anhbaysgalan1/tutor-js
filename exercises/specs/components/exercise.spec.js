import { MemoryRouter } from 'react-router-dom';
import Renderer from 'react-test-renderer';
import Exercise from '../../src/components/exercise';
import ExerciseModel from '../../src/models/exercises/exercise';
import Factory from '../factories';

jest.mock('../../../shared/src/components/html', () => ({ html }) =>
  html ? <div dangerouslySetInnerHTML={{ __html: html }} /> : null
);

describe('Exercises component', () => {
  let exercise;
  let props;

  beforeEach(() => {
    const exercises = Factory.exercisesMap();
    exercise = exercises.array[0];
    props = {
      exercises,
      match: {
        params: {
          numberWithVersion: exercise.number,
        },
      },
    };
  });

  it('renders and matches snapshot', () => {
    const ex = Renderer.create(<MemoryRouter><Exercise {...props} /></MemoryRouter>);
    expect(ex.toJSON()).toMatchSnapshot();
    ex.unmount();
  });

  it('renders with intro and a multiple questions when exercise is MC', () => {
    props.exercise = new ExerciseModel(Factory.data('Exercise', { multipart: true }));
    const ex = Renderer.create(<MemoryRouter><Exercise {...props} /></MemoryRouter>);
    expect(ex.toJSON()).toMatchSnapshot();
    ex.unmount();
  });

});
