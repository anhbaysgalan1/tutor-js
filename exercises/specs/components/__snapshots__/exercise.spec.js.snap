// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercises component renders and matches snapshot 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  Exercise was not found
</div>
`;

exports[`Exercises component renders with intro and a multiple questions when exercise is MC 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  Exercise was not found
</div>
`;
